src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1374,6 +1374,10 @@ private function registerSerializerConfiguration(array $config, ContainerBuilder
         if ($config['max_depth_handler'] ?? false) {
             $container->getDefinition('serializer.normalizer.object')->addMethodCall('setMaxDepthHandler', array(new Reference($config['max_depth_handler'])));
         }
+
+        if ($config['default_context'] ?? false) {
+            $container->getDefinition('serializer.normalizer.object')->setArgument('default_context', $config['default_context']);
```

Comment:
Actually, it would be better to use a binding: we should inject this value in all services tagged as normalizers or and encoders, and having a parameter called `$defaultContext`, even user provided ones.

There are already more services able to leverage this: https://github.com/symfony/symfony/search?q=%24defaultContext&unscoped_q=%24defaultContext
And then this setting will work out of the box in API Platform too... when we'll added support for `$defautContext` in our own normalizers.

Reply:
Is there a particular way of injecting this in every services or should I just loop and check if the service has a named argument `defaultContext`?

Reply:
You can probably reuse this: https://symfony.com/doc/current/service_container.html#binding-arguments-by-name-or-type

But you'll need a compiler pass anyway because you need to filters the service by tag. Actually, this should be added here: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php

I would do the following:

1. create a new `serializer.default_context` parameter in FrameworkBundleExtension containing the configured value
2. inject this parameter in the `$defaultContext` argument in `SerializerPass` for every encoders and normalizers.

WDYT?

Reply:
@dunglas : Why not injecting the default context in the Serializer only, which would forward it to every call as suggested in https://github.com/symfony/symfony/pull/28927#discussion_r226821641 ?

Reply:
Because we should use the `defaultContext` as much as possible: it allows to improve performance (passing the context in a stateful way increase the miss rate because of this: https://github.com/symfony/serializer/blob/d449b9a703e9c007fdaf6c9b2dd14aff2ac158b8/Normalizer/AbstractObjectNormalizer.php#L480)

Reply:
Also, it will improve the memory usage (minor), and the stack traces.